crates/fuel-core/src/query/message.rs

```diff
@@ -166,61 +165,57 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
             _ => None,
         });
 
-    let (sender, recipient, nonce, amount, data) = match receipt {
-        Some(r) => r,
-        None => return Ok(None),
+    let Some((sender, recipient, nonce, amount, data)) = receipt else {
```

Comment:
up to you, but you can tie this to the statement above with `.ok_or` if you prefer more concise statements: 

```rust
    let (sender, recipient, nonce, amount, data) = database
        .receipts(&transaction_id)?
        .into_iter()
        .find_map(|r| match r {
            Receipt::MessageOut {
                sender,
                recipient,
                nonce,
                amount,
                data,
                ..
            } if r.nonce() == Some(&desired_nonce) => {
                Some((sender, recipient, nonce, amount, data))
            }
            _ => None,
        })
        .ok_or::<StorageError>(
            anyhow::anyhow!("desired `nonce` missing in transaction receipts").into(),
        )?;
```

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2258/commits/1956a88a285406400e77de036a5890f78332016f